File: Combinations/Python/combinations.py
import sys

def main():
    if len(sys.argv) == 3:
        k = int(sys.argv[1])
        n = int(sys.argv[2])
        arr = []
        for x in range(0, n):
            arr.append(x + 1)
        helper(arr, k, n)
    else:
        print("Incorrect Input")

def combination(arr, k, n, index, data, i):
    if index == k:
        for x in range (0, k):
            print(data[x], end=' ')
        print()
        return
    if i >= n:
        return

    data[index] = arr[i]
    combination(arr, k, n, index+1, data, i+1)
    combination(arr, k, n, index, data, i+1)


def helper(arr, k, n):
    data = [0] * n
    combination(arr, k, n, 0, data, 0)


main()

File: Combinations/Ocaml/combinations.ml

let makeArray a = 
	let arr = Array.make a 0 in
		for i = 0 to a-1 do
			arr.(i) <- (i+1);
		done;
	print_string "\n";
    Array.to_list arr;
;;

let rec printList inp k =

	if inp = [] then
		print_string "\n"
	else begin
		let se = Array.of_list (List.hd (inp)) in
			for i = 0 to k-1 do
				print_int (se.(i));
				print_string " ";
			done;
		print_string "\n";
		printList (List.tl inp) k;
	end
;;

	
let rec combine k list =
    if k <= 0 then [ [] ]
    else match list with
         | [] -> []
         | h :: tl ->
            let with_h = List.map (fun l -> h :: l) (combine (k-1) tl) in
            let without_h = combine k tl in
            with_h @ without_h
;;

let k = int_of_string (Sys.argv.(1));;
let n = int_of_string (Sys.argv.(2));;
printList (combine k (makeArray n)) k;;

